src/Adapter/Product/Repository/ProductRepository.php

```diff
@@ -64,6 +84,57 @@ public function assertProductExists(ProductId $productId): void
         $this->assertObjectModelExists($productId->getValue(), 'product', ProductNotFoundException::class);
     }
 
+    /**
+     * @param ProductId[] $productIds
+     *
+     * @throws ProductNotFoundException
+     */
+    public function assertProductsExists(array $productIds): void
+    {
+        //@todo: no shop association. Should it be checked here?
+        $ids = array_map(function (ProductId $productId): int {
+            return $productId->getValue();
+        }, $productIds);
+        $ids = array_unique($ids);
+
+        $qb = $this->connection->createQueryBuilder();
+        $qb->select('COUNT(id_product) as product_count')
+            ->from($this->dbPrefix . 'product')
+            ->where('id_product IN (:productIds)')
+            ->setParameter('productIds', $ids, Connection::PARAM_INT_ARRAY);
+
+        $results = $qb->execute()->fetch();
+
+        if (!$results || (int) $results['product_count'] !== count($ids)) {
+            throw new ProductNotFoundException('Some of products does not exist');
+        }
+    }
+
+    /**
+     * @param ProductId $productId
+     * @param LanguageId $languageId
+     *
+     * @return array<string, mixed>
```

Comment:
The return format is not very clear Maybe we should add a comment for this, if I'm not mistaken it returns something like this:
```php
[
    ['id_product' => 42, 'name' => 'Product name', 'reference' => 'demo15'],
]
```